alacritty/src/display/content.rs

```diff
@@ -428,11 +429,6 @@ impl<'a> Hint<'a> {
     ///
     /// The tuple's [`bool`] will be `true` when the character is the first for this hint.
     fn advance(&mut self, viewport_start: Point, point: Point) -> Option<(char, bool)> {
-        // Check if we're within a match at all.
-        if !self.matches.advance(point) {
-            return None;
-        }
```

Comment:
Removing this seems very wrong.

Reply:
I removed it because if that function is called then it's already evaluated that we are already inside a match, first line here:

`if content.hint.as_mut().map_or(false, |hint| hint.matches.advance(cell.point)) {`
`if let Some((c, is_first)) = content.hint.as_mut().and_then(|hint| hint.advance(viewport_start, cell.point)){`

I can put it back, just to be extra safe, or if something else calls that function.